api/types/plugin/plugin_responses.go

```diff
@@ -1,16 +1,16 @@
-package types // import "github.com/docker/docker/api/types"
+package plugin
 
 import (
 	"encoding/json"
 	"fmt"
 	"sort"
 )
 
-// PluginsListResponse contains the response for the Engine API
-type PluginsListResponse []*Plugin
+// ListResponse contains the response for the Engine API
+type ListResponse []*Plugin
 
 // UnmarshalJSON implements json.Unmarshaler for PluginInterfaceType
-func (t *PluginInterfaceType) UnmarshalJSON(p []byte) error {
+func (t *InterfaceType) UnmarshalJSON(p []byte) error {
```

Comment:
What in the world?! Our Swagger definition of `PluginInterfaceType` is an object with three non-nullable properties. _But that is a lie!_ It's actually a string over the wire. We should be writing Go struct definitions in Go, and leave the Swagger spec for describing the over-the-wire API. Now's our big chance to fix the Swagger spec, to make a clean break with the api/types/plugin package and not aliasing the `"./api/types".PluginInterfaceType` and `"./api/types".PluginConfigInterface` types to the new Swagger-generated types.

Reply:
Or if our old version of go-swagger supports it, perhaps we could use https://goswagger.io/go-swagger/reference/models/schemas/#external-types and `type: string` in the Swagger spec?

Reply:
Just a note to self; 

- looks like  https://github.com/moby/moby/commit/6f7b69b63f172e9c687416ac1445b2c016f39ff6 / https://github.com/moby/moby/pull/27117 introduced this; before that patch, the type was `string` in the swagger docs
- :point_up: ironically, I once did a PR to fix the example response https://github.com/moby/moby/commit/4735c7663201ce1bf618e2aa505d7813a331be3f / https://github.com/moby/moby/pull/34250, but for some reason did not notice the fault was actually in the definition here.
- I guess  I didn't fully grasp the swagger at the time, seeing my comment on the related ticket 🙈 https://github.com/moby/moby/issues/34241